.github/workflows/node.js.yml

```diff
@@ -2,18 +2,23 @@ name: Node.js CI
 
 on:
   push:
-    branch: master
+    paths-ignore:
+      - 'docs/**'
+      - '*.md'
   pull_request:
-    branch: master
+    paths-ignore:
+      - 'docs/**'
+      - '*.md'
 
 jobs:
   build:
 
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
 
     strategy:
       matrix:
         node-version: [12.x, 14.x, 16.x]
```

Comment:
I think soon we will drop the 12.x version and we will support 14+ or maybe drop 14.x also and support 16+!!

Reply:
Yes, I think when 12.x is outdated, we can add 17.x in ci

Reply:
I don't think Koa 2 will drop any major version support. If we're following semver that would be a major break.

Reply:
Node v12.x is END-OF-LIFE now, and node v18.x will be active LTS in 2022/10/18. I think it is necessary to update ci version to consistent with nodejs releases